### PR TITLE
VIH-7793 - Time zone set to UTC and add better waits on page

### PR DESCRIPTION
**Pull request by MichaelKeaneathmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/VIH-7793

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: AcceptanceTests/AcceptanceTests.Common/Driver/Drivers/UserBrowser.cs [hunk 1]
```diff
@@ -105,15 +105,16 @@ public void Retry(Action action, int times = ActionRetries)
...
-                .Handle<StaleElementReferenceException>()
-                .WaitAndRetry(times, retryAttempt =>
-                        TimeSpan.FromSeconds(Math.Pow(2, retryAttempt)),
-                    (exception, timeSpan, context) => { NUnit.Framework.TestContext.WriteLine($"Encountered error '{exception.Message}' after {timeSpan.Seconds} seconds. Retrying..."); })
-                .Execute(action);
-        }
+        //  09/06/2021 - Keep for now as this might be meeded to be restored.
+        //private void RetryOnStaleElement(Action action, int times = ActionRetries)
+        //{
+        //    Policy
```
Comment (damien-cooke) [inline AcceptanceTests/AcceptanceTests.Common/Driver/Drivers/UserBrowser.cs:111]: I personally prefer getting rid of it and taking note of the commit ID. You can then just grab it from history. Comments tend to get left when we do stuff like this.

File: AcceptanceTests/AcceptanceTests.Common/Driver/Drivers/UserBrowser.cs [hunk 2]
```diff
@@ -183,9 +184,17 @@ public void ClickToProgress(By element, By newElement, int timeout = 20)
 
         public void Click(By element, int timeout = 20)
         {
-            RetryOnStaleElement(() => PerformClick(element, timeout));
+            // 09/06/2021 - Keep for now as this might need to be restored
+            //RetryOnStaleElement(() => PerformClick(element, timeout));
```
Comment (damien-cooke) [inline AcceptanceTests/AcceptanceTests.Common/Driver/Drivers/UserBrowser.cs:188]: Same here

File: AcceptanceTests/AcceptanceTests.Common/Driver/Drivers/UserBrowser.cs [hunk 3]
```diff
@@ -183,9 +184,17 @@ public void ClickToProgress(By element, By newElement, int timeout = 20)
...
+            NUnit.Framework.TestContext.WriteLine($"Attempting click of element {element.ToString()} on {Driver.Url} ");
+            WebDriverWait wait = new WebDriverWait(Driver, timeout: TimeSpan.FromSeconds(30))
+            {
+                PollingInterval = TimeSpan.FromSeconds(1),
```
Comment (damien-cooke) [inline AcceptanceTests/AcceptanceTests.Common/Driver/Drivers/UserBrowser.cs:192]: Is this defaulted? Are we overriding the default to make it do it every second?

File: AcceptanceTests/AcceptanceTests.Common/Driver/Drivers/UserBrowser.cs [hunk 4]
```diff
@@ -248,9 +257,13 @@ public void NavigateBack()
 
         public string TextOf(By element)
         {
-            var text = string.Empty;
-            RetryOnStaleElement(() => text = PerformGetText(element));
-            return text.IsNullOrEmpty() ? text : text.Trim();
+            WebDriverWait wait = new WebDriverWait(Driver, timeout: TimeSpan.FromSeconds(30))
+            {
+                PollingInterval = TimeSpan.FromSeconds(1),
```
Comment (damien-cooke) [inline AcceptanceTests/AcceptanceTests.Common/Driver/Drivers/UserBrowser.cs:262]: Can we make this one method that we call with different params as we have some repeated code in here

File: AcceptanceTests/AcceptanceTests.Common/Driver/Helpers/DriverExtensions.cs [hunk 5]
```diff
@@ -39,8 +39,12 @@ public static IWebElement WaitUntilElementClickable(this IWebDriver driver, By e
         {
             try
             {
-                var wait = new WebDriverWait(driver, TimeSpan.FromSeconds(timeout));
-                return wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementToBeClickable(elementLocator));
+                WebDriverWait wait = new WebDriverWait(driver, timeout: TimeSpan.FromSeconds(30))
+                {
+                    PollingInterval = TimeSpan.FromSeconds(1),
```
Comment (damien-cooke) [inline AcceptanceTests/AcceptanceTests.Common/Driver/Helpers/DriverExtensions.cs:44]: Same here for repeated code

**Review by damien-cooke — CHANGES_REQUESTED** [review]

**Review by ibthmcts — APPROVED** [review]

File: AcceptanceTests/AcceptanceTests.Common/Driver/Drivers/UserBrowser.cs (hunk 1 above)
Comment (MichaelKeaneathmcts) [reply AcceptanceTests/AcceptanceTests.Common/Driver/Drivers/UserBrowser.cs:111]: I'm not sure that I like the idea but that's just my preference.  Code has been removed and will be pushed.

File: AcceptanceTests/AcceptanceTests.Common/Driver/Drivers/UserBrowser.cs (hunk 2 above)
Comment (MichaelKeaneathmcts) [reply AcceptanceTests/AcceptanceTests.Common/Driver/Drivers/UserBrowser.cs:188]: Same as my last response

File: AcceptanceTests/AcceptanceTests.Common/Driver/Drivers/UserBrowser.cs (hunk 3 above)
Comment (MichaelKeaneathmcts) [reply AcceptanceTests/AcceptanceTests.Common/Driver/Drivers/UserBrowser.cs:192]: I'm going to convert this to a helper class as it's in two classes and may be used in more later.

File: AcceptanceTests/AcceptanceTests.Common/Driver/Drivers/UserBrowser.cs (hunk 4 above)
Comment (MichaelKeaneathmcts) [reply AcceptanceTests/AcceptanceTests.Common/Driver/Drivers/UserBrowser.cs:262]: I'm going to convert this to a helper class as it's in two classes and may be used in more later.

File: AcceptanceTests/AcceptanceTests.Common/Driver/Helpers/DriverExtensions.cs (hunk 5 above)
Comment (MichaelKeaneathmcts) [reply AcceptanceTests/AcceptanceTests.Common/Driver/Helpers/DriverExtensions.cs:44]: I'm going to convert this to a helper class as it's in two classes and may be used in more later.

Comment (ibthmcts) [reply AcceptanceTests/AcceptanceTests.Common/Driver/Helpers/DriverExtensions.cs:44]: Isn't this already centralised/ put somewhere for common access?

Comment (ibthmcts) [reply AcceptanceTests/AcceptanceTests.Common/Driver/Helpers/DriverExtensions.cs:44]: Ok.

File: AcceptanceTests/AcceptanceTests.Common/Driver/Drivers/UserBrowser.cs (hunk 3 above)
Comment (MichaelKeaneathmcts) [reply AcceptanceTests/AcceptanceTests.Common/Driver/Drivers/UserBrowser.cs]: Values are not defaulted so I will be creating a config object to read a file to set defaults like this.  VIH-7843

**Pull request merged**